copy/multiple_test.go

```diff
@@ -0,0 +1,195 @@
+				"sha256:e4c0df75810b953d6717b8f8f28298d73870e8aa2a0d5e77b8391f16fdfbbbe2",
+			},
+		},
+		{
+			id:      "platforms only in order",
+			fixture: "../manifest/fixtures/v2list.manifest.json",
+			instancePlatforms: []imgspecv1.Platform{
+				{
+					OS:           "linux",
+					Architecture: "ppc64le",
+				},
+				{
+					OS:           "linux",
+					Architecture: "s390x",
+				},
+			},
+			expected: []digest.Digest{
+				"sha256:7820f9a86d4ad15a2c4f0c0e5479298df2aa7c2f6871288e2ef8546f3e7b6783",
+				"sha256:e4c0df75810b953d6717b8f8f28298d73870e8aa2a0d5e77b8391f16fdfbbbe2",
+			},
+		},
+		{
+			id:      "platforms only out of order",
+			fixture: "../manifest/fixtures/v2list.manifest.json",
+			instancePlatforms: []imgspecv1.Platform{
+				{
+					OS:           "linux",
+					Architecture: "s390x",
+				},
+				{
+					OS:           "linux",
+					Architecture: "ppc64le",
+				},
+			},
+			expected: []digest.Digest{
+				"sha256:7820f9a86d4ad15a2c4f0c0e5479298df2aa7c2f6871288e2ef8546f3e7b6783",
+				"sha256:e4c0df75810b953d6717b8f8f28298d73870e8aa2a0d5e77b8391f16fdfbbbe2",
+			},
+		},
+		{
+			id:      "mixed without duplicates",
+			fixture: "../manifest/fixtures/v2list.manifest.json",
+			instancePlatforms: []imgspecv1.Platform{
+				{
+					OS:           "linux",
+					Architecture: "s390x",
+				},
+			},
+			instanceDigests: []digest.Digest{
+				"sha256:7820f9a86d4ad15a2c4f0c0e5479298df2aa7c2f6871288e2ef8546f3e7b6783",
+			},
+			expected: []digest.Digest{
+				"sha256:7820f9a86d4ad15a2c4f0c0e5479298df2aa7c2f6871288e2ef8546f3e7b6783",
+				"sha256:e4c0df75810b953d6717b8f8f28298d73870e8aa2a0d5e77b8391f16fdfbbbe2",
+			},
+		},
+		{
+			id:      "mixed with duplicates",
+			fixture: "../manifest/fixtures/v2list.manifest.json",
+			instancePlatforms: []imgspecv1.Platform{
+				{
+					OS:           "linux",
+					Architecture: "ppc64le",
+				},
+				{
+					OS:           "linux",
+					Architecture: "s390x",
+				},
+			},
+			instanceDigests: []digest.Digest{
+				"sha256:7820f9a86d4ad15a2c4f0c0e5479298df2aa7c2f6871288e2ef8546f3e7b6783",
+				"sha256:e4c0df75810b953d6717b8f8f28298d73870e8aa2a0d5e77b8391f16fdfbbbe2",
+			},
+			expected: []digest.Digest{
+				"sha256:7820f9a86d4ad15a2c4f0c0e5479298df2aa7c2f6871288e2ef8546f3e7b6783",
+				"sha256:e4c0df75810b953d6717b8f8f28298d73870e8aa2a0d5e77b8391f16fdfbbbe2",
+			},
+		},
+		{
+			id:      "no such platform",
+			fixture: "../manifest/fixtures/v2list.manifest.json",
+			instancePlatforms: []imgspecv1.Platform{
+				{
+					OS:           "windows",
+					Architecture: "amd64",
+				},
+				{
+					OS:           "darwin",
+					Architecture: "arm64",
+				},
+			},
+			expectedErrIncludes: "no image found in manifest list for",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.id, func(t *testing.T) {
+			listBlob, err := ioutil.ReadFile(tc.fixture)
+			if err != nil {
+				t.Fatalf("unexpected error reading fixture %q: %v", tc.fixture, err)
```

Comment:
(Absolutely non-blocking: {`assert`,`require`}`.NoError(t, err)` is a bit more concise. This works just fine as is.)